docs/adr/0035-refactor-payments-frontend.md

```diff
@@ -0,0 +1,88 @@
+
+Regarding why Next.js is favored over Gatsby.js, before completing the ADR, the subscription platform team went through a prototyping project and found the two frameworks to be remarkably similar in capabilities. However due to Next.js's popularity, use by other teams in Mozilla and a team vote, the decision was made to pick Next.js.
+
+## Pros and Cons of the Options
+
+### A. React Framework - Next.js
+
+Use a React framework, Next.js.
+
+- Good, because it provides many benefits out of the box, without compromising flexibility, that would require additional libraries otherwise. (Routing, rendering options, image optimization, etc.)
+- Good, because of its routing simplicity and flexibility
+- Good, because of its built in tooling, which reduces existing bespoke tooling currently used
+- Good, because it is easy to learn the basics
+- Good, because it's parent company has other tools that work well with Next.js and enhance its capabilities. (Turbopack, SWR, etc.)
+- Good, because of its large community support and popularity
+- Good, because it is used by other teams in Mozilla
+- Good, because it no longer uses Redux for state management
+- Bad, because it adds another layer on top of React
+- Bad, because it's new to most team members, and requires some learning to use more complicated features
+- Bad, because it diverges from tooling used elsewhere in the FxA monorepo
+
+### B. React Framework - Gatsby.js
+
+Use a React framework, Gatsby.js. (Gatsby has similar good and bad points to Next.js)
+
+- Good, because it provides many benefits out of the box, without compromising flexibility, that would require additional libraries otherwise. (Routing, rendering options, image optimization, etc.)
+- Good, because of its routing simplicity and flexibility
+- Good, because of its built in tooling, which reduces existing bespoke tooling currently used
+- Good, because it is easy to learn the basics
+- Good, because it no longer uses Redux for state management
+- Neutral, because of its large community support and popularity, but smaller compared to Next.js
+- Neutral, not currently used by other teams in Mozilla
+- Bad, because it adds another layer on top of React
+- Bad, because it's new to most team members, and requires some learning to use more complicated features
+- Bad, because it diverges from tooling used elsewhere in the FxA monorepo
+
+### C. React, Typescript, without Redux
+
+Use the existing fxa-payments-server app and only remove redux.
+
+- Good, because simplifies the app by removing complexity of using Redux
+- Good, because it would mostly use the existing technologies, and not require new learnings for the team
+- Bad, because Rescripts and other tooling make it difficult to upgrade React tooling. (See [FXA-5242](https://mozilla-hub.atlassian.net/browse/FXA-5242))
```

Comment:
rescripts was added to add or overwrite configuration from create-react-app.  CRA's the real culprit 😀